compiler.go

```diff
@@ -118,6 +118,16 @@ func (c *Compiler) AddResource(url string, r io.Reader) error {
 	return nil
 }
 
+// AddResourceJSON adds a schema from an already loaded map to the compiler.
+func (c *Compiler) AddResourceJSON(url string, doc any) error {
```

Comment:
- change `AddResource` to do `unmarshal` and call `AddResourceJson`. This avoids code repetition.
- use `interface{}` instead of any for consistency in project.

Reply:
applied